### ISSUE TITLE
.eslintignore not work if path includes special char.

### ISSUE THREAD

**Issue by xlfsummer** [eslint/eslint#10821]
**Tell us about your environment**

* **ESLint Version:**
v5.4.0

* **Node Version:**
v8.11.3

* **npm Version:**
v5.6.0

**What parser (default, Babel-ESLint, etc.) are you using?**
default

**Please show your full configuration:**
<details>
<summary>Configuration</summary>

.eslintrc
```json
{
    "rules": {
        "no-extra-semi":"error"
    }
}
```
.eslintignore
```
dist
```
</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

.\demo\dir\dist\code.js
```js
;
```

.\demo\目录\dist\code.js
```js
;
```

command:
```bash
.\node_modules\.bin\eslint .
```

**What did you expect to happen?**
no error found, because both code.js files should be ignored.

**What actually happened? Please include the actual, raw output from ESLint.**
![image](https://user-images.githubusercontent.com/20750969/44918564-dfa56880-ad6d-11e8-8e62-e95d01b36475.png)

**Here is the demo repo:**
https://github.com/xlfsummer/eslint-demo.git

**Comment by Aladdin-ADD** [comment]
thanks for the report and the demo!

I can repro it, and seems it was happening on windows, but working as expected on my ubuntu.

**Comment by aquibm** [comment]
Looks like this is because of a posix conversion in the ignore library used by [lib/ingored-paths.js](https://github.com/eslint/eslint/blob/master/lib/ignored-paths.js#L14). I've [created a PR upstream](https://github.com/kaelzhang/node-ignore/pull/46) that should resolve the issue.

If we need it resolved urgently, we could run the relative path through our [convertPathToPosix](https://github.com/eslint/eslint/blob/master/lib/util/path-utils.js#L23) function before checking if the file should be ignored.

**Comment by aquibm** [comment]
@kaicataldo 

Ignore 5.0.4 resolves this issue. [I've looked at updating the library](https://github.com/aquibm/eslint/commit/9efa19fbdb1228a18e947b00c88fd3949004f2e5) but there are some pretty massive changes between 4.x and 5.x. The library now expects all paths to be relative to some base directory and does not support `../` in the paths provided to it.

This means that eslint no longer works as expected for the following example:

```
demo/
├── dir1/
│   ├── file.js
│   ├── .eslintignore
├── dir2/
│   ├── file.js

> eslint ../
```

If the `.eslintignore` contains a rule to ignore `file.js` and we run eslint from `dir1`, we assume the base directory to be dir1 which sends relative paths from dir1 to dir2 to the ignore library causing it to throw. 

This is because we try to figure out the base directory [by comparing the cwd and the location of the .eslintignore file](https://github.com/eslint/eslint/blob/master/lib/ignored-paths.js#L266).

Should I look into working out the base directory using the files / directories that we're running eslint over rather than where it's being run from? Or is my example too contrived and unlikely to happen in the real world?
